src/fsharp/TypedTreeOps.fs

```diff
@@ -7539,7 +7539,7 @@ let rec MakeApplicationAndBetaReduceAux g (f, fty, tyargsl: TType list list, arg
       match f with 
       | Expr.TyLambda (_, tyvs, body, _, bodyty) when tyvs.Length = List.length tyargs -> 
           let tpenv = bindTypars tyvs tyargs emptyTyparInst
-          let body = remarkExpr m (instExpr g tpenv body)
+          let body = instExpr g tpenv body
           let bodyty' = instType tpenv bodyty
```

Comment:
What was the remarkExpr doing before?

Reply:
"mark" is an old name for "range" so "remarkExpr" means "re-range expression" meaning replacing the ranges with one specific range.  We use the same routine to make some other changes too (locals to compiler generated), and it is always run for inline code.

In the above line, we were needlessly and incorrectly calling "remarkExpr" even when doing simple reduction on applying generic functions (e.g. when optimizing `let id x = x in id 3` - here `id` is generic and the optimizer decides to inline it).  We shouldn't be remarking the body of `id` here.  For example consider
```fsharp
let f () = 
    let id x =
        printfn "hello"
        printfn "hello"
        x
    id 3 + id 7
```

The debug points for the `printfn` shouldn't disappear in Release code just because `id` gets inlined (that's assuming it does actually get inlined)